Sources/kai/Checker.swift

```diff
@@ -188,11 +188,14 @@ class Type: Equatable, CustomStringConvertible {
     }
 
     var isBooleanesque: Bool {
-        return !flags.union(.booleanesque).isEmpty
+        //FIXME(vdka): this is the old implementation and it ALWAYS returned true
+        //return !flags.union(.booleanesque).isEmpty
+        return flags.contains(.booleanesque) || isNumeric || isBoolean
```

Comment:
This should be right. `booleanesque` is defined as `[.numeric, .boolean]` so both numbers and actual booleans are _`booleanesque`_.
`Flag.booleanesque.rawValue` is `0b00001111`
`flags.union(.booleanesque)` is equivalent to `flags.rawValue | 0b00001111`
We then check if the result was `0` with `.isEmpty`
If the result is not `0` then we have either:
- boolean
- integer
- unsigned
- float
All of these are convertible to booleans through truncation currently